packages/aragon-wrapper/src/index.js

```diff
@@ -642,17 +642,33 @@ export default class Aragon {
       throw new Error(`No method named ${methodName} on ${destination}`)
     }
 
+    const methodABI = app.abi.find(
+      (method) => method.name === methodName
+    )
+    if (!methodABI) {
+      throw new Error(`${methodName} not found on ABI for ${destination}`)
+    }
+
+    let transactionOptions = {
+      gasPrice: defaultGasPrice,
+    }
+
+    // If an extra parameter has been provided, it is the transaction options
+    if (methodABI.inputs.length + 1 == params.length && typeof params[params.length - 1] === 'object') {
+      const options = params.pop()
+      transactionOptions = { ...transactionOptions, ...options }
+    }
+
+    if (methodABI.inputs.length != params.length) {
+      throw new Error(`Incorrect number of parameters for function ${methodABI.name}. Expected ${methodABI.input.length}, provided ${params.length}`)
```

Comment:
If the above conditional fails on the typeof statement I could make a case for having a separate error message but it's probably not necessary.